amida-tech/amida-messaging-microservice
# Feature/js threaded replies

#### What's this PR do?
Implements backend message threading.

#### Related JIRA tickets:
https://jira.amida-tech.com/browse/SER-14

#### How should this be manually tested?
First, make sure all tests pass. Then, manually run some replies with cURL. Ensure that the `originalMessageId` for messages in a thread is always set to the message ID of the message owned by the original sender. Using this one message as the root will allow a client to query and piece together all messages in a thread using parent message IDs plus this root ID.

Definitely let me know if the message tree logic seems correct and usable for Indaba or another messaging app.
